File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/CleanupPhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

class CleanupPhase extends BenchmarkPhase {

    CleanupPhase(Connection connection) {
        super(connection);
    }

    @Override
    protected Status execute() {
        return getConnection().cleanup();
    }
}

File: cpp/src/benchmark/phase/CleanupPhase.cpp
#include "CleanupPhase.h"
#include <benchmark/Benchmark.h>

#include <utility>

namespace Observatory {

CleanupPhase::CleanupPhase(Benchmark &benchmark) :
        BenchmarkPhase(benchmark) {}

const char* CleanupPhase::getName() {
    return "CleanupPhase";
}

Status CleanupPhase::execute() {
    return getBenchmark().cleanup();
}

}